bundles/org.eclipse.orion.client.ui/web/orion/widgets/projects/RunBar.js

```diff
@@ -216,13 +223,20 @@ define([
 					var dropdownMenuItemSpan = lib.$(".dropdownMenuItem", createNewItem); //$NON-NLS-0$
 					dropdownMenuItemSpan.classList.add("addNewMenuItem"); //$NON-NLS-0$
 					
-					var defaultDeployCommand = this._projectCommands.getDeployProjectCommands(this._commandRegistry)[0];
-					if (defaultDeployCommand) {
-						this._commandRegistry.registerCommandContribution(createNewItem.id, defaultDeployCommand.id, 1); //$NON-NLS-0$
+					var defaultDeployCommands = this._projectCommands.getDeployProjectCommands(this._commandRegistry);
+					if (defaultDeployCommands) {
+						for (var i = 0; i < defaultDeployCommands.length; i++) {
+							this._commandRegistry.registerCommandContribution(createNewItem.id, defaultDeployCommands[i].id, i + 1); //$NON-NLS-0$
+						}
 						domNodeWrapperList = [];
 						this._commandRegistry.renderCommands(createNewItem.id, dropdownMenuItemSpan, this._project, this, "button", null, domNodeWrapperList); //$NON-NLS-0$
-						domNodeWrapperList[0].domNode.textContent = "+"; //$NON-NLS-0$
-						this._setNodeTooltip(domNodeWrapperList[0].domNode, messages["createNewTooltip"]); //$NON-NLS-0$
+						for (var i = 0; i < defaultDeployCommands.length; i++) {
+							domNodeWrapperList[i].domNode.textContent = defaultDeployCommands[i].name;
+							this._setNodeTooltip(domNodeWrapperList[i].domNode, defaultDeployCommands[i].tooltip);
+						}
+						if (defaultDeployCommands.length === 1) {
+							domNodeWrapperList[0].domNode.textContent = "+";
```

Comment:
It will be a "dropdown inside a dropdown", which I think it's not good.

Another solution can be: If there is more than one deploy services and the user clicks "+", multiple buttons (like the current ones in this PR) shows and replace the "+" button.

Example:

| Runbar |
| --- |
| Config 1 |
| Config 2 |
| [+] |

After the user clicks the [+] button:

| Runbar |
| --- |
| Config 1 |
| Config 2 |
| [Service 1] [Service 2] |